Fix shutdown hang with >= 8 -addnodes set

TheBlueMatt opened the pull request:
We previously would block waiting for a CSemaphoreGrant, when we
did not need to.

laanwj commented:
utACK https://github.com/bitcoin/bitcoin/pull/9953/commits/569b6eec8a640581f78432688567f04e2a4f3056

TheBlueMatt committed:
Fix shutdown hang with >= 8 -addnodes set

We previously would block waiting for a CSemaphoreGrant in
ThreadOpenAddedConnections, when we did not need to. This would
block as the posts in CConnman shutdown were both to the wrong
semaphore and in the wrong location.

TheBlueMatt commented:
Changed the fix to post to semAddnode instead of TryAcquire.

TheBlueMatt committed:
Add missing braces in semaphore posts in net

gmaxwell reviewed (approved):
utACK. (will test) Thanks.

theuni commented:
utACK 819b513a5415d1669b5440e214862cda6c2261f8

jonasschnelli commented:
utACK 819b513a5415d1669b5440e214862cda6c2261f8

jonasschnelli commented:
Should fix: #9950

gmaxwell reviewed (approved):
ACK.
